lib/tether/tether.go

```diff
@@ -252,6 +258,46 @@ func (t *tether) setMounts() error {
 			return fmt.Errorf("unsupported volume mount type for %s: %s", k, v.Source.Scheme)
 		}
 	}
+	return t.populateVolumes()
+}
+
+func (t *tether) populateVolumes() error {
+	defer trace.End(trace.Begin(fmt.Sprintf("populateVolumes")))
+	// skip if no mounts present
+	if len(t.config.Mounts) == 0 {
+		return nil
+	}
+	// skip if this was done before
+	if _, err := os.Stat(mountsCopied); err == nil {
```

Comment:
This is fine for now, but somewhat violates our goal of idempotent behaviour - if Reload is called and a new volume has been configured (or the CopyMode has changed) then we will not populate newly added volumes.

If we add more complex Copy behaviours (rsync for example) then this would be more simply phrased as a checking function per Copy type that takes a source and dest and returns whether or not to perform the copy, or just copies if needed.

Please open a follow up issue as a child of #4774 to revisit this in a tech debt, with a reference in #696 as that's where we will likely need the revised behaviour.